### PR TITLE
Add dateShiftScope with resource, file and folder

### PR DESCRIPTION
**Pull request by BoyaWu10**
This pull request is to enable date shift with scope: resource, file and folder.

### CODE REVIEW

File: src/Fhir.Anonymizer.Core/Utility/DateTimeUtility.cs [hunk 1]
```diff
@@ -155,12 +155,15 @@ private static bool IndicateAgeOverThreshold(GroupCollection groups)
             return age > s_ageThreshold; 
         }
 
-        private static int GetDateShiftValue(ElementNode node, string dateShiftKey)
+        private static int GetDateShiftValue(ElementNode node, string dateShiftKey, string dateShiftPrefix)
         {
-            string resourceId = TryGetResourceId(node);
+            if (string.IsNullOrEmpty(dateShiftPrefix))
+            {
+                dateShiftPrefix = TryGetResourceId(node);
```
Comment (moria97) [inline src/Fhir.Anonymizer.Core/Utility/DateTimeUtility.cs:162]: Would we use resource Id as dateshift scope by default?

File: src/Fhir.Anonymizer.Tool/FilesAnonymizerForJsonFormatResource.cs [hunk 2]
```diff
@@ -124,5 +122,23 @@ private string GetResourceOutputFileName(string fileName, string inputFolder, st
 
             return Path.Combine(outputFolder, partialFilename);
         }
+
+        private AnonymizerEngine CreateAnonymizerEngineForFile(string filePath)
```
Comment (moria97) [inline src/Fhir.Anonymizer.Tool/FilesAnonymizerForJsonFormatResource.cs:126]: Shall we add a static CreateForResourceFile(stirng configPath, string resourceFileName) function in AnonymizerEngine to reuse this logic?

File: src/Fhir.Anonymizer.Core/Utility/DateTimeUtility.cs (hunk 1 above)
Comment (BoyaWu10) [reply src/Fhir.Anonymizer.Core/Utility/DateTimeUtility.cs:162]: Yes, the default date shift scope is resource. It uses resource Id as dateShiftPrefix to compute shifting amount.

File: src/Fhir.Anonymizer.Tool/FilesAnonymizerForJsonFormatResource.cs (hunk 2 above)
Comment (BoyaWu10) [reply src/Fhir.Anonymizer.Tool/FilesAnonymizerForJsonFormatResource.cs]: Agree! It is added in the update.

File: src/Fhir.Anonymizer.Core/Utility/DateTimeUtility.cs [hunk 3]
```diff
@@ -119,7 +119,7 @@ public static void ShiftDateTimeAndInstantNode(ElementNode node, string dateShif
             var matchedGroups = s_dateTimeRegex.Match(node.Value.ToString()).Groups;
             if (matchedGroups[s_dayIndex].Captures.Any() && !IndicateAgeOverThreshold(matchedGroups))
             {
-                int offset = GetDateShiftValue(node, dateShiftKey);
+                int offset = GetDateShiftValue(node, dateShiftKey, dateShiftPrefix);
```
Comment (skywalker-wu) [inline src/Fhir.Anonymizer.Core/Utility/DateTimeUtility.cs:122]: Perhaps dateShiftKeyPrefix?

Comment (BoyaWu10) [reply src/Fhir.Anonymizer.Core/Utility/DateTimeUtility.cs]: Yes, it is fixed in the update.

**Pull request merged**